cobbler/actions/buildiso/standalone_buildiso.py

```diff
@@ -0,0 +1,267 @@
+    """
+    This class contains all functionality related to building self-contained installation images.
+    """
+
+    def _validate_standalone_args(self, distro_name: str, source: str):
+        """
+        If building standalone, we only want --distro and --profiles (optional), systems are disallowed
+
+        :param distro_name: The name of the distribution we want to install.
+        :param source: The source which we copy to the ISO as an installation base.
+        """
+        if not distro_name:
+            raise ValueError("When building a standalone ISO, you must specify a --distro")
+        if not os.path.exists(source):
+            raise ValueError("The specified source \"%s\" does not exist!" % source)
+
+        # Ensure all profiles specified are children of the distro
+        if self.profiles:
+            base_distro = self.api.find_distro(name=distro_name)
+            orphan_profiles = [profile for profile in self.profiles if profile not in base_distro.children]
+            if len(orphan_profiles) > 0:
+                raise ValueError("When building a standalone ISO, all --profiles must be under --distro")
+
+    def _sync_airgapped_repos(self, airgapped: bool, repo_names_to_copy: dict):
+        """
+        Syncs all repositories locally available into the image if the is supposed to be airgapped.
+
+        :param airgapped: If false this method is doing nothing.
+        :param repo_names_to_copy: The names of the repositories which should be included.
+        :raises RuntimeError: In case the rsync of the repository was not successful.
+        """
+        if airgapped:
+            # copy any repos found in profiles or systems to the iso build
+            repodir = os.path.abspath(os.path.join(self.isolinuxdir, "..", "repo_mirror"))
+            if not os.path.exists(repodir):
+                os.makedirs(repodir)
+
+            for repo_name in repo_names_to_copy:
+                self.logger.info(" - copying repo '%s' for airgapped ISO", repo_name)
+                rsync_successful = utils.rsync_files(
+                    repo_names_to_copy[repo_name],
+                    os.path.join(repodir, repo_name),
+                    "--exclude=TRANS.TBL --exclude=cache/ --no-g", quiet=True)
+                if not rsync_successful:
+                    raise RuntimeError("rsync of repo \"%s\" failed" % repo_name)
+
+    def _validate_standalone_filesource(self, filesource: str, distro) -> str:
+        """
+        Validate that the path to the installation sources is making sense. If they do then normalize the path and
+        return it.
+
+        :param filesource: The path to the installation sources.
+        :param distro: The distribution of which the kernel is used for booting the image.
+        :raises ValueError: In case the installation source was not found.
+        :return: Normalized filesource which points to the absolute path of the source tree.
+        """
+        if filesource is None:
```

Comment:
Okay the problem here is "bigger" then I thought. The logic in the method you linked (`_validate_standalone_filesource`) is currently fully overwriting the `filesource` in case we also hand a distro to the function. This is always the case when building a standalone ISO. Thus the whole parameter get's discarded in the first place which questions the documentation of the method and its usage.

Reply:
Fixed via rearranging the checks